api/proto/scheduler/scheduler.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+enum RunnerType {
+  DOCKER = 0;
+}
+
+message Action {
+  string action_id = 1;
+  RunnerType runner_type = 2;
+  optional string image = 3;
+  repeated string commands = 4;
+}
+
+message RegisteredAgent {
+  string hostname = 1;
+  int32 cpu = 2;
```

Comment:
This is really weird for me to include Health in the register request. Maybe if you really want to do it you can re-use the HealthStatus message inside this interface

Reply:
I agree. The registering request could be empty and the scheduler would assign an ID to the registered agent and send it back to the agent as part of the register response.

Reply:
I still find this uncomfortable to need two call to register since i can't use the agent while I don't have the information about it's capacity.

Reply:
I see. You could consider that before getting the first status update from the agent, the scheduler can't schedule anything there. It's registered but passive. Or you could include a status message as part of the registration, both solutions are fine with me.

Reply:
we decided (with the agent team) to give the health at the register